File: src/client/commands/free.ts
/**
 * Free Command Module
 * 
 * Tests the free endpoint to demonstrate what users get WITHOUT paying.
 * Creates clear contrast with premium content for educational purposes.
 */

import axios from 'axios';
import type { CLICommand, CommandContext } from '../types/commands';
import { displayError } from '../utils/display';
import { logger } from '../../shared/utils/logger';

/**
 * Free endpoint test command implementation
 */
export const freeCommand: CLICommand = {
  name: 'free',
  aliases: [],
  description: 'Test free endpoint for comparison (no payment required)',
  usage: 'free',
  
  async execute(args: string[], context: CommandContext): Promise<void> {
    try {
      logger.header('Free Content Test', 'Testing free endpoint access');
      
      logger.flow('request', { 
        action: 'Accessing free content', 
        endpoint: '/free',
        payment: 'NOT REQUIRED' 
      });
      
      // Simple axios request - no payment interceptor needed
      const response = await axios.get('http://localhost:3000/free', {
        timeout: 10000
      });
      
      if (response.data) {
        logger.ui('\n📖 FREE CONTENT ACCESSED');
        logger.ui('═══════════════════════════════════════════════════════════');
        
        if (response.data.message) {
          logger.ui(`📢 ${response.data.message}`);
        }
        
        if (response.data.subtitle) {
          logger.ui(`   ${response.data.subtitle}`);
        }
        
        const data = response.data.data;
        if (data) {
          // Basic Info
          if (data.basicInfo) {
            logger.ui('\n📋 Basic Information:');
            logger.ui(`   Service: ${data.basicInfo.service}`);
            logger.ui(`   Version: ${data.basicInfo.version}`);
            logger.ui(`   Access Level: ${data.basicInfo.accessLevel}`);
          }
          
          // Free Features
          if (data.freeFeatures && data.freeFeatures.length > 0) {
            logger.ui('\n🆓 What You Get For Free:');
            data.freeFeatures.forEach((feature: string) => {
              logger.ui(`   ${feature}`);
            });
          }
          
          // Limitations
          if (data.limitations) {
            logger.ui('\n⚠️  Limitations of Free Tier:');
            logger.ui(`   Update Frequency: ${data.limitations.updateFrequency}`);
            logger.ui(`   Data Accuracy: ${data.limitations.dataAccuracy}`);
            logger.ui(`   API Calls/Hour: ${data.limitations.apiCallsPerHour}`);
            logger.ui(`   Support: ${data.limitations.supportLevel}`);
            logger.ui(`   Advanced Features: ${data.limitations.advancedFeatures}`);
          }
          
          // Upgrade Information
          if (data.upgradeInfo) {
            logger.ui('\n💡 Want More?');
            logger.ui(`   ${data.upgradeInfo.note}`);
            logger.ui(`   ${data.upgradeInfo.upgrade}`);
            logger.ui(`   Benefits: ${data.upgradeInfo.benefits}`);
          }
        }
        
        logger.ui('\n🆓 This content was FREE - no payment required!');
        logger.ui('   Compare this with the "test" command to see premium features.');
      }
      
      logger.separator();
      
    } catch (error: any) {
      if (error.code === 'ECONNREFUSED') {
        displayError('Cannot connect to server at http://localhost:3000');
        logger.ui('💡 Make sure the server is running: npm run dev:server');
      } else {
        displayError('Error during free endpoint test', error);
      }
    }
  }
}; 

File: python/src/client/commands/free.py
"""
Free Command Module

Accesses the free endpoint without requiring payment.
Demonstrates basic API access and response handling.
"""

import asyncio
import aiohttp
from typing import Dict, Any
from src.shared.utils.logger import logger
from src.shared.config import config as shared_config


async def free_command(args: list) -> None:
    """
    Free command implementation
    
    Args:
        args: Command arguments (unused)
    """
    try:
        # Log free endpoint access attempt
        logger.flow('free_endpoint', {
            'action': 'Accessing free endpoint',
            'timestamp': '2025-06-20T03:24:45.110Z'
        })

        # Get server configuration
        server_config = shared_config.get_server_config("python")
        base_url = f"http://{server_config['host']}:{server_config['port']}"
        
        # Create HTTP client and access free endpoint
        logger.ui('\n🔓 Accessing free endpoint...')
        
        async with aiohttp.ClientSession() as session:
            async with session.get(f"{base_url}/free", timeout=aiohttp.ClientTimeout(total=10)) as response:
                if response.status != 200:
                    logger.error(f'Server error: {response.status}')
                    logger.ui(f'💡 Make sure the server is running: npm run py:server')
                    return
                
                response_data = await response.json()
                
                # Validate response format
                if not response_data or not isinstance(response_data, dict):
                    logger.error('Invalid response format from free endpoint')
                    return

                # Display response
                if response_data.get('message'):
                    logger.ui(f"\n📢 {response_data['message']}")
                if response_data.get('subtitle'):
                    logger.ui(f"   {response_data['subtitle']}")

                # Display free content details
                if response_data.get('data'):
                    data = response_data['data']
                    
                    # Basic info
                    if data.get('basicInfo'):
                        info = data['basicInfo']
                        logger.ui(f"\n📋 Basic Information:")
                        logger.ui(f"   Service: {info.get('service', 'N/A')}")
                        logger.ui(f"   Version: {info.get('version', 'N/A')}")
                        logger.ui(f"   Access Level: {info.get('accessLevel', 'N/A')}")
                    
                    # Free features
                    if data.get('freeFeatures'):
                        logger.ui(f"\n✅ Free Features:")
                        for feature in data['freeFeatures']:
                            logger.ui(f"   {feature}")
                    
                    # Limitations
                    if data.get('limitations'):
                        limits = data['limitations']
                        logger.ui(f"\n⚠️  Limitations:")
                        logger.ui(f"   Update Frequency: {limits.get('updateFrequency', 'N/A')}")
                        logger.ui(f"   Data Accuracy: {limits.get('dataAccuracy', 'N/A')}")
                        logger.ui(f"   API Calls/Hour: {limits.get('apiCallsPerHour', 'N/A')}")
                        logger.ui(f"   Support Level: {limits.get('supportLevel', 'N/A')}")
                    
                    # Upgrade info
                    if data.get('upgradeInfo'):
                        upgrade = data['upgradeInfo']
                        logger.ui(f"\n💡 {upgrade.get('note', 'Want more features?')}")
                        logger.ui(f"   {upgrade.get('upgrade', 'Try premium tiers')}")
                        logger.ui(f"   {upgrade.get('benefits', 'Unlock premium features')}")

                # Log successful access
                logger.flow('free_success', {
                    'action': 'Accessed free endpoint',
                    'status': response.status,
                    'timestamp': '2025-06-20T03:24:45.110Z'
                })
                
    except asyncio.TimeoutError:
        logger.error('Request timeout - server may be down')
        logger.ui('💡 Make sure the server is running: npm run py:server')
    except aiohttp.ClientConnectorError:
        logger.error('Network error: Cannot connect to server')
        logger.ui('💡 Make sure the server is running: npm run py:server')
    except Exception as e:
        logger.error('Failed to access free endpoint', e)
        logger.ui('💡 Make sure the server is running: npm run py:server')


# Command registration
COMMANDS = {
    'free': {
        'function': free_command,
        'description': 'Access the free endpoint without payment',
        'usage': 'free'
    }
} 